community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/commands/expressions/DistanceFunction.scala

```diff
@@ -59,49 +57,99 @@ case class DistanceFunction(p1: Expression, p2: Expression) extends Expression {
   override def toString = "Distance(" + p1 + ", " + p2 + ")"
 }
 
-trait DistanceCalculator {
-  def isDefinedAt(p1: PointValue, p2: PointValue): Boolean
-
-  def calculateDistance(p1: PointValue, p2: PointValue): Double
-
-  def apply(p1: PointValue, p2: PointValue): Option[Double] =
-    if (isDefinedAt(p1, p2))
-      Some(calculateDistance(p1, p2))
-    else
-      None
+trait DistanceCalculator extends PartialFunction[(PointValue, PointValue), Double] {
```

Comment:
These calculators are more general than the DistanceFunction itself and so should reside somewhere else. Current thinking is to move them to the CRS.